pkg/apis/multiclusterservice/v1/types.go

```diff
@@ -0,0 +1,56 @@
+package v1
+
+import (
+	"net"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//MultiClusterService describes the information of service that is available across
+//clusters. A shared informer running in the lighthouse control plane will listen for
+//service creation in every cluster that is part of the control plane and whenever a
+//service is created a MultiClusterService will be created. Lighthouse will
+//distribute the MultiClusterService to every other cluster that is the part of the
+//control plane. When the service is updated/deleted lighthouse ensures that the
+//updated state is reflected in every cluster.
+type MultiClusterService struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              MultiClusterServiceSpec `json:"spec"`
+}
+
+//MultiClusterServiceSpec provides the specification of a MultiClusterService
+type MultiClusterServiceSpec struct {
+	//List of service information across clusters. This field is not optional.
+	Items []ClusterServiceInfo `json:"clusterServiceInfo"`
+}
+
+//ClusterServiceInfo provides information about service in each cluster.
+type ClusterServiceInfo struct {
+	//The unique identity of the cluster where the service is available. This field is
+	//not optional.
+	ClusterID string `json:"clusterID"`
+
+	//The DNS zone in which the service is a part of. This field is not optional.
+	ClusterDomain string `json:"clusterDomain"`
+
+	//The cluster IP of the service running in the cluster that is identified by the
+	//ClusterID field. This field is not optional.
+	ServiceIP net.IP `json:"serviceIP"`
+
+	//+optional
+	//The port in which the service listens at.This is an optional field.
+	Port int `json:"port,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//MultiClusterServiceList is a list of multiclusterservice objects.
+type MultiClusterServiceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []MultiClusterService `json:"items"`
+}
```

Comment:
General comment:

This CRD is too verbose. Can't you already have a list of a dictionary where key is the Domain and the value is the IP/PORT?

Anyways the controller is the one that takes those decisions, there is no need to provide such a structured CRD IMHO

Reply:
I was under the impression this is the general practice as I modeled  it based on the other CRD's. Doesn't it make it easier to test create /delete the CRD's using kubectl?

Reply:
I'm not sure I understand the advantage you say this gives. Could you explain me with an example or differently?